libtiledbvcf/src/stats/variant_stats.cc

```diff
@@ -806,20 +809,91 @@ std::string VariantStats::get_uri(const std::string& root_uri, bool relative) {
+                current_sample_offset - *(sample_offsets_point - 1);
+            pos_equal = *(pos_buffer_point - 1) == pos_;
+            end_diff = (static_cast<int64_t>(end) - *(end_buffer_point - 1));
+            end_out_of_order = (end_diff < 0);
+            sample_diff = strncmp(
+                sample_buffer_.data() + *(sample_offsets_point - 1),
+                sample_.data(),
+                std::min<uint32_t>(last_sample_length, sample_.size()));
+            sample_out_of_order =
+                (sample_diff > 0 ||
+                 (sample_diff == 0 && last_sample_length > sample_.size()));
+            sample_equal =
+                (sample_diff == 0 && last_sample_length == sample_.length());
+            not_underrunning = end_buffer_point - 1 > end_buffer_.begin();
+            return static_cast<bool>(  // need to swap ends
+                (not_underrunning && pos_equal && sample_equal &&
+                 end_out_of_order) ||
+                // need to swap samples
+                (not_underrunning && pos_equal && sample_out_of_order));
+          };
+          while (/*cells are out of order*/ out_of_order()) {
+            // decrement iterators by one cell
+            contig_offsets_point--;
+            contig_buffer_point =
+                contig_buffer_.begin() + *contig_offsets_point;
+            *contig_offsets_point += contig_.length();
+            pos_buffer_point--;
+            sample_offsets_point--;
+            current_sample_offset = *sample_offsets_point;
+            sample_buffer_point =
+                sample_buffer_.begin() + *sample_offsets_point;
+            *sample_offsets_point += sample_.length();
+            allele_offsets_point--;
+            allele_buffer_point =
+                allele_buffer_.begin() + *allele_offsets_point;
+            *allele_offsets_point += allele.length();
+            ac_buffer_point--;
+            an_buffer_point--;
+            n_hom_buffer_point--;
+            end_buffer_point--;
+          }
+        }
+        max_length_buffer_.push_back(value.max_length);
+        end_buffer_.insert(end_buffer_point, value.end);
       }
+      contig_offsets_.insert(
+          contig_offsets_point, contig_buffer_point - contig_buffer_.begin());
+      contig_buffer_.insert(
+          contig_buffer_point, contig_.begin(), contig_.end());
+      pos_buffer_.insert(pos_buffer_point, pos_);
+      sample_offsets_.insert(
+          sample_offsets_point, sample_buffer_point - sample_buffer_.begin());
+      sample_buffer_.insert(
+          sample_buffer_point, sample_.begin(), sample_.end());
+      allele_offsets_.insert(
+          allele_offsets_point, allele_buffer_point - allele_buffer_.begin());
+      allele_buffer_.insert(allele_buffer_point, allele.begin(), allele.end());
+      ac_buffer_.insert(ac_buffer_point, value.ac);
+      an_buffer_.insert(an_buffer_point, value.an);
+      n_hom_buffer_.insert(n_hom_buffer_point, value.n_hom);
```

Comment:
This looks overly complicated. Let's review in person.

Reply:
after discussion: needs more documentation